src/EnergyPlus/ReportCoilSelection.cc

```diff
@@ -1044,7 +1044,48 @@ ReportCoilSelection::setCoilAirFlow(
 }
 
 void
-ReportCoilSelection::setCoilWaterFlow(
+ReportCoilSelection::setCoilWaterFlowNodeNums(
+	std::string const & coilName, // user-defined name of the coil
+	std::string const & coilType, // idf input object class name of coil
+	Real64 const waterVdot, // plant fluid flow rate in m3/s
+	bool const isAutoSized, // true if water flow was autosized
+	int const inletNodeNum, // coil chw inlet node num
+	int const outletNodeNum, // coil chw outlet node num
+	int const plantLoopNum // plant loop structure index
+)
+{
+	int index = getIndexForOrCreateDataObjFromCoilName( coilName,coilType );
+	auto & c( coilSelectionDataObjs[ index ] );
+	if ( ( DataSizing::NumPltSizInput > 0 ) && ( inletNodeNum > 0 ) && ( outletNodeNum > 0 ) ) {
+		bool errorsfound = false;
+		c->pltSizNum = DataPlant::MyPlantSizingIndex( "water coil", coilName, inletNodeNum, outletNodeNum, errorsfound );
+	} else {
+		c->pltSizNum = -999;
+	}
+	c->waterLoopNum = plantLoopNum;
+
+	if ( c->waterLoopNum > 0 && c->pltSizNum > 0 ) {
+		c->plantLoopName = DataPlant::PlantLoop( c->waterLoopNum ).Name;
+		if ( DataSizing::PlantSizData( c->pltSizNum  ).LoopType != DataSizing::SteamLoop ) {
+			c->rhoFluid = FluidProperties::GetDensityGlycol( DataPlant::PlantLoop( c->waterLoopNum ).FluidName, DataGlobals::InitConvTemp, DataPlant::PlantLoop( c->waterLoopNum ).FluidIndex, "ReportCoilSelection::setCoilWaterFlow" );
```

Comment:
Thanks for being explicit on the scope of these functions and not using using statements!